basic/src/test/java/org/jboss/hal/testsuite/test/runtime/batch/BatchManagementTestCase.java

```diff
@@ -171,22 +175,37 @@ public void startJobTest() throws Exception {
 
     @Test
     public void stopJobTest() throws Exception {
-        deployBatchApplication(1000);
+        final long stoppingInterval = 5000;
+        deployBatchApplicationWithBatchlet(stoppingInterval);
         long executionId = batchOps.startJob(deploymentName, jobOneFileName);
         page.navigate2jobs();
         ResourceTableRowFragment jobRow = page.selectRowForJob(jobOneFileName);
         assertEquals(STARTED, page.getBatchStatusFromRow(jobRow));
-
         page.clickButton("Stop");
         assertEquals(STOPPING, page.getBatchStatusFromRow(jobRow));
         String actualStatus = batchOps.getExecution(deploymentName, jobOneName, executionId).batchStatus;
         assertTrue("We expected STOPPING or STOPPED but got '" + actualStatus + "'.",
                 actualStatus.equals(STOPPING) || actualStatus.equals(STOPPED));
+        assertJobHasStoppedOrFail(jobOneFileName, stoppingInterval);
+    }
+
+    private void assertJobHasStoppedOrFail(String jobName, long timeoutInMillis) throws InterruptedException {
+        long startTime = System.currentTimeMillis();
```

Comment:
Since `startTime` is changing I would suggest to rename it to `currentTime` to make the loop condition more clear.